partitura/performance.py

```diff
@@ -137,18 +141,31 @@ def from_note_array(cls, note_array,
         Note that this property does not include non-note information (i.e.
         controls such as sustain pedal).
         """
-        if not 'id' in note_array.dtype.names:
+        if 'id' not in note_array.dtype.names:
             n_ids = ['n{0}'.format(i) for i in range(len(note_array))]
         else:
             n_ids = note_array['id']
 
+        if 'track' not in note_array.dtype.names:
+            track = np.zeros(len(note_array), dtype=int)
+        else:
+            track = note_array['track']
+
+        if 'channel' not in note_array.dtype.names:
+            channel = np.ones(len(note_array), dtype=int)
+        else:
+            channel = note_array['channel']
+
         notes = []
-        for nid, note in zip(n_ids, note_array):
+        for nid, note, track, channel in zip(n_ids, note_array,
+                                             track, channel):
```

Comment:
it's a bit confusing to reuse the variable names `track`/`channel` in the for loop.

Reply:
Fixed in a4fcd88.